src/components/SEO.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react'
+import Helmet from 'react-helmet'
+import PropTypes from 'prop-types'
+import { StaticQuery, graphql } from 'gatsby'
+
+const query = graphql`
+  query GetSiteMetadata {
+    site {
+      siteMetadata {
+        title
+        author
+        description
+        siteUrl
+        social {
+          twitter
+        }
+      }
+    }
+  }
+`
+
+function SEO({ meta, image, title, description, slug }) {
+  return (
+    <StaticQuery
+      query={query}
+      render={data => {
+        const { siteMetadata } = data.site
+        const metaDescription = description || siteMetadata.description
+        const metaImage = image ? `${siteMetadata.siteUrl}/${image}` : null
+        const url = `${siteMetadata.siteUrl}${slug}`
+        return (
+          <Helmet
+            htmlAttributes={{ lang: 'en' }}
+            {...(title
+              ? {
+                  titleTemplate: `%s - ${siteMetadata.title}`,
```

Comment:
react-helmet has some weird behavior here. If you _don't_ pass a title (e.g. for the index route, we want the title to just be the site name) it renders empty/null.

So we need a check here to only use titleTemplate if we have a title passed in.